rir/src/compiler/pir/instruction.h

```diff
@@ -240,6 +240,7 @@ class Instruction : public Value {
     bool usesAreOnly(BB*, std::unordered_set<Tag>);
     bool usesDoNotInclude(BB*, std::unordered_set<Tag>);
     bool unused();
+    virtual bool isDebug() const { return false; }
```

Comment:
not neccessary. instead, set effects to `Effects::Any()` to ensure they are not removed.